### PR TITLE
test: Add Ubuntu 16.04 Image

### PR DESCRIPTION
**Pull request by stefwalter**
Prepare testing on Ubuntu 16.04
- [x] Build the image
- [x] Make sure pbuilder works
- [x] Add to testing cycle
- [x] Disable kubernetes tests
- [ ] Update tests
- [ ] Make NetworkManager work

### CODE REVIEW

**Comment by stefwalter** [comment]
@larsu I'm stuck on this, when running ```vm-create -v ubuntu-1604```

```
I: Obtaining the cached apt archive contents
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libpcp3-dev
E: Unable to locate package libpcp-import1-dev
E: Unable to locate package libpcp-pmda3-dev
E: Unable to locate package xmlto
E: Unable to locate package nodejs-legacy
E: Unable to locate package npm
I: Copying back the cached apt archive contents
```

**Comment by stefwalter** [comment]
I've gotten further than the above, now failing on missing dh.

**Comment by stefwalter** [comment]
Stuck at: ```make: dh: Command not found```

**Comment by stefwalter** [comment]
Solved the build issues. Will upload an new image shortly.

**Comment by stefwalter** [comment]
Needs lots more work sorting out individual tests.

**Comment by stefwalter** [comment]
Blocked on #5227

**Comment by stefwalter** [comment]
@mvollmer I'm having trouble with the ```check-storage-msdos``` test. It fails when looking for ```Free Space for Primary Partitions``` on line 57 of testDosParts.

File: pkg/storaged/details.js [hunk 1]
```diff
@@ -269,15 +269,24 @@
...
                                           return block_ptable.CreatePartition(start, vals.size, "", "", { });
-                                      else
+                                      } else if (block_ptable.CreatePartitionAndFormat) {
                                           return block_ptable.CreatePartitionAndFormat (start, vals.size, "", "", { },
                                                                                         vals.type, options);
-                                  } else
+                                      } else {
+                                          return block_ptable.CreatePartition(start, vals.size, "", "", { })
+                                              .then(function(partition) {
+                                                  return block.Format(vals.type, options).then(function() {
```
Comment (mvollmer) [inline pkg/storaged/details.js]: This should call format on client.blocks[partition], not on block which refers to the whole disk at this point.

Comment (stefwalter) [inline pkg/storaged/details.js]: Thanks. That did the trick.

**Comment by stefwalter** [comment]
@mvollmer So I made it all the way through 6 tests ... but now I'm stuck on:

```
+ journalctl
Journal extracted to TestStorage-testLuks-10.111.122.243-FAIL.log
Traceback (most recent call last):
  File "./check-storage-luks", line 59, in testLuks
    self.wait_in_storaged_configuration(mount_point_secret)
  File "/data/src/cockpit/test/verify/storagelib.py", line 220, in wait_in_storaged_configuration
    wait(lambda: mount_point in self.machine.execute("%s dump | grep Configuration" % self.storagectl_cmd))
  File "/data/src/cockpit/test/common/testlib.py", line 1118, in wait
    raise Error(msg or "Condition did not become true.")
Error: Condition did not become true.
```

**Comment by mvollmer** [comment]
> but now I'm stuck on:

The storaged CreatePartitionAndFormat method has a "config-item" option.  With UDisks2, one has to call AddConfigurationItem explicitly after a format.

See https://github.com/storaged-project/storaged/commit/5125c6c67d804b1b376b22438bcf0d6f8df41bfa

So we have to add a call to AddConfigurationItem, and pass a value for fsname.

**Comment by stefwalter** [comment]
> So we have to add a call to AddConfigurationItem, and pass a value for fsname.

I think you mean "fstab" ... but yes makes sense.

**Comment by stefwalter** [comment]
@mvollmer So I've solved completing the configuration item data client side. However I've run into a very different issue:

```
-> wait('ph_is_present("#content .list-group li:nth-child(1) .btn-group")')
<- True
-> eval('ph_click("#content .list-group li:nth-child(1) .btn-group a:contains(\\"Delete\\")",true)')
<- u''
-> wait('ph_is_present("#dialog")')
<- True
-> wait('ph_is_visible("#dialog")')
<- True
-> eval('ph_click("#dialog [data-action=\\"apply\\"]",false)')
<- u''
-> wait('!ph_is_present("#dialog")')
<- True
-> wait('ph_in_text("#content","Free Space")')
<- True
-> wait('!ph_in_text("#content","ENCRYPTED")')
<- raise timeout
-> show('TestStorage-testLuks-FAIL.png')
Wrote TestStorage-testLuks-FAIL.png
<- None
+ journalctl
Journal extracted to TestStorage-testLuks-10.111.126.111-FAIL.log
Traceback (most recent call last):
  File "./check-storage-luks", line 93, in testLuks
    b.wait_not_in_text("#content", "ENCRYPTED")
  File "/data/src/cockpit/test/common/testlib.py", line 244, in wait_not_in_text
    return self.wait_js_func('!ph_in_text', selector, text)
  File "/data/src/cockpit/test/common/testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/data/src/cockpit/test/common/testlib.py", line 692, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/data/src/cockpit/test/common/testlib.py", line 715, in _invoke
    raise Error(res['error'])
Error: timeout
```

it doesn't seem that deleting a partition actually happens properly. After a delete of a partition that has a luks mapper device on it ... I can see the changes in parted, but the kernel doesn't update /dev and udisks2 doesn't update.

@mvollmer Could you have a look?

File: pkg/storaged/details.js [hunk 2]
```diff
@@ -129,6 +166,28 @@
...
+            function create_and_format_partition(block_ptable, start, size, type, options, config_items) {
+                return block_ptable.CreatePartition(start, size, "", "", { })
+                    .then(function(partition) {
+                        var block = client.blocks[partition];
+                        var promises = config_items.map(function(item) {
+                            item = complete_configuration_item(block, item);
+                            return block.AddConfigurationItem(item, { });
+                        });
+                        return cockpit.all(promises).then(function() {
+                            return block.Format(type, options).then(function() {
```
Comment (mvollmer) [inline pkg/storaged/details.js]: Format will change the `block.IdUUID` property and we want to put the new value into fstab and crypttab, not the old one. Also Format might create a new cleartext block device and we need to fall back to that if there is no UUID.  (Hmm, I wonder if udisks/storaged has the same bug, actually.)

So AddConfigurationItem needs to happen after the Format.

**Comment by mvollmer** [comment]
> Could you have a look?

Yes.  Would you like me to finish this entirely?

**Comment by stefwalter** [comment]
> Would you like me to finish this entirely? 

Sure. I'll push what I have.

**Comment by stefwalter** [comment]
Blocked on #5232 and #5236

**Comment by mvollmer** [comment]
> it doesn't seem that deleting a partition actually happens properly.

Exactly.  UDisks2 doesn't have the 'teardown' option and will not lock the Luks device.  Consequently, the partition is in use (by device mapper) when UDisks2 tries to delete it with parted.

Parted puts the system into a inconsistent state and merely warns about it.  UDisks2 swallows that warning because parted hasn't actually failed:

```
# parted /dev/sda --script "rm 1"
Error: Partition(s) 1 on /dev/sda have been written, but we have been unable to inform the kernel of the change, probably because it/they are in use.  As a result, the old partition(s) will remain in use.  You should reboot now before making further changes.
root@unassigned-hostname:~# echo $?
0
```

The test could lock the Luks device before removing, but UDisks2 would still fail to remove the fstab and crypttab entries.  I guess we can skip that part of the test.

**Comment by mvollmer** [comment]
> Parted puts the system into a inconsistent state and merely warns about it. UDisks2 swallows that warning because parted hasn't actually failed.

The UI could prevent this by disabling "Delete" on a unlocked luks device when talking to UDisks2.

**Comment by stefwalter** [comment]
> The UI could prevent this by disabling "Delete" on a unlocked luks device when talking to UDisks2.

Yes, that would be the best way. Disable it and put a tooltip saying why it's disabled.

**Comment by mvollmer** [comment]
> Yes, that would be the best way. Disable it and put a tooltip saying why it's disabled.

I'll do that. Not sure whether we can have a tooltip in the menu... but with #5097 we definitely can.

File: pkg/storaged/client.js [hunk 3]
```diff
@@ -333,83 +333,113 @@
...
-         * try them first.
+    function init_manager() {
+        /* Storaged 2.6 and later uses the UDisks2 API names, but try the
+         * older storaged API first as a fallback.
          */
 
-        STORAGED_SERVICE =   "org.freedesktop.UDisks2";
-        STORAGED_OPATH_PFX = "/org/freedesktop/UDisks2";
-        STORAGED_IFACE_PFX = "org.freedesktop.UDisks2";
+        var storaged_service = "org.freedesktop.Storaged";
```
Comment (mvollmer) [inline pkg/storaged/client.js]: It's `org.storaged.Storaged`

**Comment by mvollmer** [comment]
Continued in #5285

**Pull request closed without merging**